### PR TITLE
DOC: many small changes for Leipzig course

### PR DESCRIPTION
**Pull request by GaelVaroquaux**
Includes:

- Fixing rst formatting
- Ordering enumerations to have most important items for application
  first
- Fixing text and includes to relate to current examples

Any chance that this could be merged this morning?

### CODE REVIEW

File: doc/manipulating_images/data_preparation.rst [hunk 1]
```diff
@@ -10,30 +10,53 @@ Data preparation: loading and basic signal extraction
...
+data loaded in memory. In the example above, the
+function :func:`smooth_img` returns a NiftiImage object, which can then
+be readily passed to other nilearn functions.
+
+In nilearn, we often use the term *"niimg"* as abbreviation that denotes
+either a file name or a `NiftiImage object
+<http://nipy.org/nibabel/nibabel_images.html>`_. 
+
+Niimgs can be 3D or 4D. A 4D niimg can be **a list of file names**, if
+these contain 3D information.
```
Comment (AlexandreAbraham) [inline doc/manipulating_images/data_preparation.rst]: Should we also say 4D with 1 scan?

Comment (GaelVaroquaux) [inline doc/manipulating_images/data_preparation.rst]: > Should we also say 4D with 1 scan?

You mean in the 3D part? Sound reasonnable. Do we enforce this
systematically?

Comment (AlexandreAbraham) [inline doc/manipulating_images/data_preparation.rst]: I think so, I have to check. Otherwise, LGTM. You can merge it and I will fix this détail later.

File: doc/manipulating_images/data_preparation.rst [hunk 2]
```diff
@@ -10,30 +10,53 @@ Data preparation: loading and basic signal extraction
...
 |
 
-.. topic:: **File names as arguments**
+.. currentmodule:: nilearn.image
 
-   Nilearn functions and objects accept file names as arguments::
+File names or image objects
+=============================
+
+File names and objects for 3D and 4D images
```
Comment (banilo) [inline doc/manipulating_images/data_preparation.rst]: nitpick 1: perhaps something with "time series" instead of 4D as this is more a computerscience-vision and potentially unclear to newcomers

Comment (GaelVaroquaux) [inline doc/manipulating_images/data_preparation.rst]: > nitpick 1: perhaps something with "time series" instead of 4D as this
> is more a computerscience-vision and potentially unclear to newcomers

Fair enough: can you suggest a wording?

File: doc/manipulating_images/data_preparation.rst (hunk 1 above)
Comment (GaelVaroquaux) [inline doc/manipulating_images/data_preparation.rst]: Can you have a look, and correct in master (or with a short PR) after this merge. Thx!

File: doc/manipulating_images/data_preparation.rst [hunk 3]
```diff
@@ -10,30 +10,53 @@ Data preparation: loading and basic signal extraction
...
+  Note that the resulting is a 4D image.
 
-    >>> smoothed_imgs = image.smooth_img('~/t_map*.nii')  # doctest: +SKIP
+* The ``~`` symbol is also expanded to the user home folder::
 
+    >>> smoothed_imgs = image.smooth_img('~/t_map001.nii')  # doctest: +SKIP
+
+  Using ``~`` rather than specifying the details of the path is good
+  practice, as it will make it more likely that your script work on
+  multiple different computers.
```
Comment (banilo) [inline doc/manipulating_images/data_preparation.rst]: leave out "multiple"

File: doc/manipulating_images/data_preparation.rst [hunk 4]
```diff
@@ -144,6 +169,9 @@ Computing the mask
     :doc:`plot_nifti_simple.py <../auto_examples/04_manipulating_images/plot_nifti_simple>`.
 
 
+Visualizing the mask computed
```
Comment (banilo) [inline doc/manipulating_images/data_preparation.rst]: reconsider grammar

File: doc/manipulating_images/data_preparation.rst [hunk 5]
```diff
@@ -214,95 +261,41 @@ Common data preparation steps: resampling, smoothing, filtering
...
+  standardized (scaled to unit variance). 
 
-- Confound removal. Two ways of removing confounds are provided. Any linear
-  trend can be removed by activating the `detrend` option.
+- **Frequency filtering**. Low-pass and high-pass filters can be used to
+  remove artifacts. Parameters: ``high_pass`` and ``low_pass``, specified
+  in Hz (note that you must specific the sampling rate in seconds with
+  the ``t_r`` parameter.
+
+- **Confound removal**. Two ways of removing confounds are provided. Any linear
```
Comment (banilo) [inline doc/manipulating_images/data_preparation.rst]: Make clearer that the first way relies on prespecified behavioral/other indicators while the detrending does not require user-input to perform a notion of confound removal

File: doc/manipulating_images/data_preparation.rst (hunk 3 above)
Comment (GaelVaroquaux) [inline doc/manipulating_images/data_preparation.rst]: > > +  multiple different computers.

> leave out "multiple"

Indeed.

File: doc/manipulating_images/data_preparation.rst (hunk 4 above)
Comment (GaelVaroquaux) [inline doc/manipulating_images/data_preparation.rst]: > > +Visualizing the mask computed

> reconsider grammar

"Visualizing the computed mask"?

File: doc/manipulating_images/manipulating_images.rst [hunk 6]
```diff
@@ -59,50 +59,59 @@ follows::
...
-Note that you can copy that folder to another user's computers to avoid
-the initial dataset download on the first fetching call.
+     * the folder specified by `data_dir` parameter in the fetching function
+     * the global environment variable `NILEARN_SHARED_DATA`
+     * the user environment variable `NILEARN_DATA`
+     * the `nilearn_data` folder in the user home folder
 
+    The two different environment variables (NILEARN_SHARED_DATA and
+    NILEARN_DATA) are provided for multi-user systems, to distinguish a
+    global dataset repository that may be read-only at the user-level.
```
Comment (banilo) [inline doc/manipulating_images/manipulating_images.rst]: global may not be clear to everybody here

File: doc/manipulating_images/manipulating_images.rst [hunk 7]
```diff
@@ -299,6 +310,67 @@ set up your own data preparation procedure:
...
+
+.. image:: ../auto_examples/04_manipulating_images/images/sphx_glr_plot_resample_to_template_001.png
+    :target: ../auto_examples/04_manipulating_images/plot_resample_to_template.html
+    :scale: 55%
+
+Resampling specific target affine, shape, or resolution
+--------------------------------------------------------
+
+The resampling procedure takes as input the
+`target_affine` to resample (resize, rotate...) images in order to match
```
Comment (banilo) [inline doc/manipulating_images/manipulating_images.rst]: somewhat unefficient sentence strucutre

File: doc/manipulating_images/data_preparation.rst (hunk 5 above)
Comment (GaelVaroquaux) [inline doc/manipulating_images/data_preparation.rst]: > Make clearer that the first way relies on prespecified behavioral/other
> indicators while the detrending does not require user-input to perform
> a notion of confound removal

Addressed

File: doc/manipulating_images/manipulating_images.rst (hunk 6 above)
Comment (GaelVaroquaux) [inline doc/manipulating_images/manipulating_images.rst]: > global may not be clear to everybody here

OK but:

- This is not a change introduced by this PR
- This paragraph is for advanced users

So, skipping for now.

File: examples/04_manipulating_images/plot_mask_computation.py [hunk 8]
```diff
@@ -93,6 +89,9 @@
 trended_data = trended.fit_transform(nyu_img)
 detrended_data = detrended.fit_transform(nyu_img)
 
+# The timeseries are numpy arrays, so we can manipulate them with numpy
```
Comment (banilo) [inline examples/04_manipulating_images/plot_mask_computation.py]: recursive statement

File: doc/manipulating_images/data_preparation.rst (hunk 2 above)
Comment (banilo) [inline doc/manipulating_images/data_preparation.rst]: "time series of 3D images (i.e., 4D)"

File: doc/manipulating_images/manipulating_images.rst (hunk 7 above)
Comment (GaelVaroquaux) [inline doc/manipulating_images/manipulating_images.rst]: > somewhat unefficient sentence strucutre

Agreed, but this sentence was already in the doc, I just moved it.
Skipping in the interest of time.

File: doc/manipulating_images/data_preparation.rst (hunk 4 above)
Comment (banilo) [inline doc/manipulating_images/data_preparation.rst]: sounds good

File: doc/manipulating_images/data_preparation.rst [hunk 9]
```diff
@@ -214,122 +261,91 @@ Common data preparation steps: resampling, smoothing, filtering
...
-    image using a reference image with :func:`nilearn.image.resample_to_img`.
+- **Standardization**. Parameter ``standardize``: Signals can be
+  standardized (scaled to unit variance). 
 
-.. image:: ../auto_examples/04_manipulating_images/images/sphx_glr_plot_resample_to_template_001.png
-    :target: ../auto_examples/04_manipulating_images/plot_resample_to_template.html
-    :scale: 33%
+- **Frequency filtering**. Low-pass and high-pass filters can be used to
+  remove artifacts. Parameters: ``high_pass`` and ``low_pass``, specified
+  in Hz (note that you must specific the sampling rate in seconds with
```
Comment (banilo) [inline doc/manipulating_images/data_preparation.rst]: perhaps give an example with concrete numbers, people might worry about how to specify that exactly

File: doc/manipulating_images/data_preparation.rst (hunk 2 above)
Comment (GaelVaroquaux) [inline doc/manipulating_images/data_preparation.rst]: > "time series of 3D images (i.e., 4D)"

It makes a horribly long title though. Are you sure that it is that
important?

Comment (GaelVaroquaux) [inline doc/manipulating_images/data_preparation.rst]: > > "time series of 3D images (i.e., 4D)"

> It makes a horribly long title though. Are you sure that it is that
> important?

I am detailing this wording below, in the text, rather than in the title.

Comment (GaelVaroquaux) [inline doc/manipulating_images/data_preparation.rst]: Thanks a lot for the feedback. I have taken it in account. I am merging,
in the interest of short iterations (and because my time is up).
